tests/filecheck/mlir-conversion/with-mlir/ptr_loop_folding.mlir

```diff
@@ -0,0 +1,79 @@
+// RUN: xdsl-opt -p convert-memref-to-ptr,convert-ptr-type-offsets,mlir-opt[scf-for-loop-canonicalization,scf-for-loop-range-folding,scf-for-loop-canonicalization],scf-for-loop-flatten,mlir-opt[scf-for-loop-canonicalization,scf-for-loop-range-folding,scf-for-loop-canonicalization] --split-input-file %s | filecheck %s
```

Comment:
How much are we missing in xDSL from the mlir-opt pipeline here? If it's not a lot, I'd much rather have this logic in xDSL.

Reply:
I believe both `scf-for-loop-canonicalization` and `scf-for-loop-range-folding` are missing. Do you think it's worth porting them?

Reply:
scf-for-loop-range-folding is already in, and was less buggy than the MLIR one until recently, where I upstreamed a bug fix to MLIR after noticing the difference with xDSL :)

Reply:
How about we bench it first? I'm just curious to see if it makes any improvements. Plus, if there is no speed increase, maybe the port won't be worth it.

Reply:
Oh, right, there is. But I think it's only for the riscv loops. Do we want a general scf version?

Reply:
I can guarantee speed improvements, and register pressure improvements in the final assembly, if that's what you mean

Reply:
yep, I think it would be worth having an scf version for this

Reply:
ok, I'll port them to xdsl

Reply:
Do we now have everything we need?